float32 u
float32 v
int32   octave
float32 angle
float32 response

